### PR TITLE
Fix TS and ESLint in Examples

### PR DESCRIPTION
**Pull request by marmor157**
## Summary
This PR fixes most of the TS and ESLint errors in Examples.

### CODE REVIEW

File: app/src/examples/LayoutAnimations/NestedNativeStacksWithLayout.tsx [hunk 1]
```diff
@@ -3,10 +3,15 @@ import { Button, StyleSheet, View } from 'react-native';
...
+import { useNavigationBuilder } from '@react-navigation/native';
 
 const Stack = createNativeStackNavigator();
 
-function First({ navigation }: { navigation: any }) {
+function First({
+  navigation,
+}: {
+  navigation: ReturnType<typeof useNavigationBuilder>['navigation'];
+}) {
```
Comment (tomekzaw) [inline app/src/examples/LayoutAnimations/NestedNativeStacksWithLayout.tsx:14]: There must be a better way to do this, maybe we need to pass some type to `createNativeStackNavigator`?

File: app/src/examples/PinExample.tsx [hunk 2]
```diff
@@ -53,14 +53,21 @@ function NumberDisplay({ number }: { number: Animated.SharedValue<number> }) {
...
 }
 
-function Digit({ digit }: { digit: Animated.SharedValue<number> }) {
+function Digit({
+  number,
+  index,
+}: {
+  number: Animated.SharedValue<number>;
+  index: number;
+}) {
```
Comment (tomekzaw) [inline app/src/examples/PinExample.tsx:69]: Move to `DigitProps`?

File: app/src/examples/SharedElementTransitions/FlatList.tsx [hunk 3]
```diff
@@ -6,16 +6,34 @@ import {
...
+  item: T;
+  parentItem: ListRenderItemInfo<T>;
+};
+
+type ParamList = {
+  Screen1: {};
+  Screen2: { sharedTransitionTag: string };
+};
+
+const Stack = createNativeStackNavigator<ParamList>();
```
Comment (tomekzaw) [inline app/src/examples/SharedElementTransitions/FlatList.tsx:35]: add a newline here
```suggestion
const Stack = createNativeStackNavigator<ParamList>();

```

File: app/src/examples/SharedStyleExample.tsx [hunk 4]
```diff
@@ -12,7 +12,7 @@ export default function SharedStyleExample() {
 
   const [blueCounter, setBlueCounter] = useState<number>(0);
   const [greenCounter, setGreenCounter] = useState<number>(0);
-  const [itemList, setItemList] = useState<any>([]);
+  const [itemList, setItemList] = useState<JSX.Element[]>([]);
```
Comment (tomekzaw) [inline app/src/examples/SharedStyleExample.tsx:15]: is `JSX.Element` indeed a correct choice here? why not `React.Node`?

File: app/src/examples/LiquidSwipe/LiquidSwipe.tsx [hunk 5]
```diff
@@ -71,12 +71,11 @@ export default function LiquidSwipe() {
...
-      let goBack: number;
+      let goBack: boolean;
       if (isBack.value) {
-        goBack = progress.value > 0.5 ? 1 : 0;
+        goBack = progress.value > 0.5;
       } else {
-        // TODO: want to use a boolean here
-        goBack = progress.value > 0.2 ? 1 : 0;
+        goBack = progress.value > 0.2;
       }
```
Comment (tomekzaw) [inline app/src/examples/LiquidSwipe/LiquidSwipe.tsx:79]: what about ...?
```ts
const goBack = isBack.value ? (progress.value > 0.5) : (progress.value > 0.2);
```

File: app/src/examples/SharedElementTransitions/Card.tsx [hunk 6]
```diff
@@ -1,23 +1,39 @@
...
 
+type ParamList = {
+  Screen1: {};
+  Screen2: {
+    title: string;
+    sharedTransitionTag: string;
+  };
+};
+
 const photo = require('./assets/image.jpg');
```
Comment (tomekzaw) [inline app/src/examples/SharedElementTransitions/Card.tsx:18]: add a newline here

**Review by tomekzaw — CHANGES_REQUESTED** [review]
thanks a lot for fixing this!

File: app/src/examples/LayoutAnimations/NestedNativeStacksWithLayout.tsx (hunk 1 above)
Comment (marmor157) [reply app/src/examples/LayoutAnimations/NestedNativeStacksWithLayout.tsx:14]: You can see that I've started from here and it was before I've learned how StackNavigator works 😆

File: app/src/examples/SharedElementTransitions/Card.tsx (hunk 6 above)
Comment (marmor157) [reply app/src/examples/SharedElementTransitions/Card.tsx:18]: [820fbee](https://github.com/software-mansion/react-native-reanimated/pull/5361/commits/820fbeed0ecb53bb5c88e4774da13f165c51ac0f)

File: app/src/examples/SharedElementTransitions/FlatList.tsx (hunk 3 above)
Comment (marmor157) [reply app/src/examples/SharedElementTransitions/FlatList.tsx:35]: [db5e86a](https://github.com/software-mansion/react-native-reanimated/pull/5361/commits/db5e86a6988501c42c8f3b2302cc1a0bd4297d10)

File: app/src/examples/SharedStyleExample.tsx (hunk 4 above)
Comment (marmor157) [reply app/src/examples/SharedStyleExample.tsx]: [b3f0654](https://github.com/software-mansion/react-native-reanimated/pull/5361/commits/b3f065496ac61dc94b104961fa1f5d2098f2953a)

File: app/src/examples/PinExample.tsx (hunk 2 above)
Comment (marmor157) [reply app/src/examples/PinExample.tsx]: [ab7a8a0](https://github.com/software-mansion/react-native-reanimated/pull/5361/commits/ab7a8a01963612cf2a92ef6d94c1224ecdb90403)

File: app/src/examples/LayoutAnimations/NestedNativeStacksWithLayout.tsx (hunk 1 above)
Comment (marmor157) [reply app/src/examples/LayoutAnimations/NestedNativeStacksWithLayout.tsx]: [08842e9](https://github.com/software-mansion/react-native-reanimated/pull/5361/commits/08842e9ecd7250acf96afc002d3bd9ec388fa950)

File: app/src/examples/IPodExample.tsx [hunk 7]
```diff
@@ -181,6 +161,57 @@ export default function IPodExample() {
...
+
+function SongCover({
+  artist,
+  song,
+  index,
+  position,
+  itemTotalSize,
+  borderMargin,
+}: SongCoverProps) {
+  const uas = useAnimatedStyle(() => {
```
Comment (tjzel) [inline app/src/examples/IPodExample.tsx:181]: It's just ugly here like this 😅 
```suggestion
  const animatedStyle = useAnimatedStyle(() => {
```

File: app/src/examples/IPodExample.tsx [hunk 8]
```diff
@@ -181,6 +161,57 @@ export default function IPodExample() {
...
+    if (index === 0) {
+      style.marginLeft = borderMargin;
+    } else if (index === data.length - 1) {
+      style.marginRight = borderMargin;
+    }
+    return style;
+  });
+
+  return (
+    <Animated.View style={[styles.item, uas]}>
```
Comment (tjzel) [inline app/src/examples/IPodExample.tsx:205]:
```suggestion
    <Animated.View style={[styles.item, animatedStyle]}>
```

File: app/src/examples/SharedElementTransitions/ImageStack.tsx [hunk 9]
```diff
@@ -12,13 +11,22 @@ import {
...
 const florence = require('./assets/florence.jpg');
 const countryside = require('./assets/countryside.jpg');
 const dawn = require('./assets/dawn.jpg');
 
-const Stack = createNativeStackNavigator();
+type ParamList = {
+  Screen1?: {};
+  Screen2?: {};
+  Screen3: {
+    image: ImageSourcePropType | SharedValue<ImageSourcePropType>;
```
Comment (tjzel) [inline app/src/examples/SharedElementTransitions/ImageStack.tsx:24]:
```suggestion
    image: ImageSourcePropType;
```

File: app/src/examples/SharedElementTransitions/Card.tsx [hunk 10]
```diff
@@ -1,23 +1,40 @@
...
-import { ParamListBase } from '@react-navigation/native';
 import {
   createNativeStackNavigator,
+  NativeStackNavigationProp,
   NativeStackScreenProps,
 } from '@react-navigation/native-stack';
 import Animated from 'react-native-reanimated';
 
+type ParamList = {
+  Screen1?: {};
```
Comment (tjzel) [inline app/src/examples/SharedElementTransitions/Card.tsx:11]: Any reason why shouldn't it be

`Screen1: object`
?

File: app/src/examples/SharedElementTransitions/FlatList.tsx [hunk 11]
```diff
@@ -6,16 +6,35 @@ import {
...
+  title: string;
+};
+
+type ListItemWithParent<T> = {
+  item: T;
+  parentItem: ListRenderItemInfo<T>;
+};
+
+type ParamList = {
+  Screen1?: {};
```
Comment (tjzel) [inline app/src/examples/SharedElementTransitions/FlatList.tsx:31]: same here

File: app/src/examples/SharedElementTransitions/ImageStack.tsx [hunk 12]
```diff
@@ -12,13 +11,22 @@ import {
...
-import Animated from 'react-native-reanimated';
+import Animated, { SharedValue } from 'react-native-reanimated';
 
 const florence = require('./assets/florence.jpg');
 const countryside = require('./assets/countryside.jpg');
 const dawn = require('./assets/dawn.jpg');
 
-const Stack = createNativeStackNavigator();
+type ParamList = {
+  Screen1?: {};
```
Comment (tjzel) [inline app/src/examples/SharedElementTransitions/ImageStack.tsx:21]: same here

File: app/src/examples/SharedElementTransitions/ImageStack.tsx [hunk 13]
```diff
@@ -12,13 +11,22 @@ import {
...
+  Screen2?: {};
```
Comment (tjzel) [inline app/src/examples/SharedElementTransitions/ImageStack.tsx:22]: same here

File: app/src/examples/SharedElementTransitions/Modals.tsx [hunk 14]
```diff
@@ -17,17 +16,30 @@ import {
   PanGestureHandlerGestureEvent,
 } from 'react-native-gesture-handler';
 
+type ParamList = {
+  Screen1?: {};
```
Comment (tjzel) [inline app/src/examples/SharedElementTransitions/Modals.tsx:20]: same here

File: app/src/examples/SharedElementTransitions/Profiles.tsx [hunk 15]
```diff
@@ -31,10 +33,16 @@ const leavesBackground = require('./assets/nature/leaves.jpg');
 const windowWidth = Dimensions.get('window').width;
 const windowHeight = Dimensions.get('window').height;
 
+type Item = {
+  id: string;
+  image: ImageSourcePropType | SharedValue<ImageSourcePropType>;
```
Comment (tjzel) [inline app/src/examples/SharedElementTransitions/Profiles.tsx:38]:
```suggestion
  image: ImageSourcePropType;
```

**Review by tjzel — CHANGES_REQUESTED** [review]
Only minor things that I'd like changed.

File: app/src/examples/SharedElementTransitions/Card.tsx (hunk 10 above)
Comment (marmor157) [reply app/src/examples/SharedElementTransitions/Card.tsx:11]: Well, actually it should be `Record<PropertyKey, never>`.  But I agree, `object` would be better than `{}`

File: app/src/examples/SharedElementTransitions/FlatList.tsx [hunk 16]
```diff
@@ -78,10 +103,18 @@ function Screen1({ navigation }: NativeStackScreenProps<ParamListBase>) {
...
+  route,
+  navigation,
+}: NativeStackScreenProps<ParamList, 'Screen2'>) {
   return (
     <View style={styles.flexOne}>
-      <Button title="go back" onPress={() => navigation.navigate('Screen1')} />
+      <Button
+        title="go back"
+        onPress={() =>
+          navigation.navigate('Screen1', { sharedTransitionTag: '' })
```
Comment (tomekzaw) [inline app/src/examples/SharedElementTransitions/FlatList.tsx:115]: is it possible to avoid passing `sharedTransitionTag` here?

**Pull request merged**